src/vm/eventpipeprovider.cpp

```diff
@@ -230,11 +230,24 @@ void EventPipeProvider::AddEvent(EventPipeEvent &event)
 
         const COUNT_T BUFFER_SIZE = dstBuffer.GetCount() + 1;
         buffer.AllocThrows(BUFFER_SIZE);
+        BOOL isQuotedValue = false;
+        COUNT_T j = 0;
         for (COUNT_T i = 0; i < BUFFER_SIZE; ++i)
-            buffer[i] = (dstBuffer[i] == '=' || dstBuffer[i] == ';') ? '\0' : dstBuffer[i];
+        {
+            if (dstBuffer[i] == '"')
```

Comment:
Do we need to handle escaped quotes?

Reply:
WE talked about this yesterday, but it requires a slightly larger change for 3.0. Do we have any customer scenarios that justify it? It honestly would just need an extra bool flag for a condition (is last character an escape and are we in a quoted string, if so, copy it into the buffer).

Reply:
In the interest of keeping the change minimal and targeted, I'm going to say that filter string values sent via EventPipe should be wrapped in quotes (`"`) if they contain `=` or `;` characters and that they **cannot** have nested quotes (`"`) in them, e.g., `name:keywords:level:key="value;=";key2=foobar;key3=5`.  

This will need to be documented and will be EventPipe specific since this code _only_ affects EventPipe calling `EventSource.OnEventCommand`.

Therefore, the following char array input to EventPipe via the Diagnostics IPC server
```
Microsoft-DotNETCore-Runtime:0:0:CustomFilter="my;custom=encoding";key=value
```
would result in the following array of `char` for the filter
```
CustomFilter\0my;custom=encoding\0key\0value\0
```

Reply:
Is your plan to add the documentation to the existing documentation (over in the PerfView repo)? Or are you planning to document something else? I'd prefer to see this change include some kind of link to the document you intend to change (hopefully in the near) future, or some kind of comment including what we are doing to the filter data.

Reply:
I'm trying to decide where to document it.  I'm currently leaning towards the dotnet/diagnostics repo.  Where in PerfView are you thinking this would be documented?

Reply:
I meant alongside the TraceEvent library documentation (.md files under https://github.com/microsoft/perfview/tree/master/src/TraceEvent/EventPipe). Putting it under diagnostics repo sounds good to me too.

Reply:
I think documenting it in the IPC protocol document would be a reasonable place. It is now part of the behavior of the CollectTracing RPC call.

Reply:
And as a side note, this issue is revealing the weaknesses in using IPC text encoding of what is nominally a Dictionary<string,string> (at least that is how EventListener API defines it). If we ever do a V3 of this RPC call, we should make use of our binary serialization format rather than trying to nest a text encoding format inside that the runtime is forced to parse. Adding the quote handling feels like a fine call for the current situation though.